subt/artf_reporter.py

```diff
@@ -37,21 +38,74 @@ def __init__(self, config, bus):
         self.artf_xyz = []
         self.artf_xyz_accumulated = []  # items are [type, position, source, flag]
                                         # position = [x, y, z], flag(scored) = True/False/None
+        self.verbose = False
 
     def publish_artf(self, artf_xyz):
         count = 0
-        for artf_type, pos, src, scored in artf_xyz:
+        for artf_type, pos, src, scored in self.group_artf_for_report(artf_xyz):
             if scored is None:
-                if count == 0:
+                if count == 0 and self.verbose:
                     print(self.time, "DETECTED:")
                 count += 1
                 ix, iy, iz = pos
-                print(" ", artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
+                if self.verbose:
+                    print(" ", artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
                 s = '%s %.2f %.2f %.2f\n' % (artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
                 self.publish('artf_cmd', bytes('artf ' + s, encoding='ascii'))
-        if count > 0:
+        if count > 0 and self.verbose:
             print('report completed')
 
+    def group_artf_for_report(self, artf_xyz):
+        """
+        There are cases when two or more robots discover artefact independently and they do not
+        have the base answer yet. Never-the-less only one representative (team identical) should
+        be selected and reported.
+        """
+        # first sort all known artifacts into three groups
+        scored_true, scored_false, scored_unknown = [], [], []
+        for item in artf_xyz:
+            artf_type, pos, src, scored = item
+            if scored is True:
+                scored_true.append(item)
+            elif scored is False:
+                scored_false.append(item)
+            else:
+                assert scored is None, scored
+                scored_unknown.append(item)
+
+        # remove all unknown by already successfully scored artifacts
+        tmp = []
+        for item in scored_unknown:
+            artf_type, pos, _, _ = item
+            for artf_type2, pos2, _, _ in scored_true:
+                # TODO do we want to filter out also other artifacts of different type (i.e. probably wrongly recognized?)
```

Comment:
+1. It is unlikely that there is a drill half a meter from a successfully detected backpack.